packages/cli/src/commands/__tests__/execHandler.test.js

```diff
@@ -0,0 +1,217 @@
+import { vol } from 'memfs'
+
+import { runScriptFunction as mockRunScriptFunction } from '../../lib/exec'
+import { generatePrismaClient as mockGeneratePrismaClient } from '../../lib/generatePrismaClient'
+import { handler } from '../execHandler'
+
+jest.mock('fs', () => require('memfs').fs)
+jest.mock('../../lib/exec')
+jest.mock('../../lib/generatePrismaClient', () => {
+  return {
+    generatePrismaClient: jest.fn().mockResolvedValue(true),
+  }
+})
+jest.mock('listr2', () => {
+  return {
+    // Return a constructor function, since we're calling `new` on Listr
+    Listr: jest.fn().mockImplementation((tasks) => {
+      return {
+        run: async () => {
+          for (const task of tasks) {
+            const skip =
+              typeof task.skip === 'function' ? task.skip : () => task.skip
+
+            if (!skip()) {
+              await task.task()
+            }
+          }
+        },
+      }
+    }),
+  }
+})
+
+// since process.exit() is used in execHandler for control flow, need to simulate it
+// being called without actually letting it kill the process
+const mockExit = jest.spyOn(process, 'exit').mockImplementation((number) => {
+  throw new Error(`process.exit(${number})`)
+})
```

Comment:
I'm wondering if there is a better way to handle checking the case where `handler` exits early via `process.exit(1)`. This code attempts to simulate that happening, without actually letting process.exit kill the test process itself, but it feels a little funny to suddenly have the handler throw when it otherwise wouldn't. Any ideas? 

The verification code looks like this:
```javascript
      await expect(handler({ name: 'foo', prisma: true })).rejects.toThrow(
        'process.exit(1)'
      )
```

Reply:
Another option would be to place an unreachable `return` statement following `process.exit(1)`, so that when combined with a stubbed implementation of process.exit the return statement would cause control flow to go back to the unit test.

```javascript
  if (targetMatch.length !== 1) {
    console.error(...) 

    printAvailableScriptsToConsole()
    process.exit(1)
    return // unreachable except when testing
  }
```

```javascript
const mockExit = jest.spyOn(process, 'exit').mockImplementation((number) => {})

...

await handler(...)
expect(mockExit).toHaveBeenCalledWith(1)
```

This makes the verification code a little bit cleaner, and doesn't cause process.exit to throw where it otherwise wouldn't have.

But I don't like the idea of needing to include that unreachable statement in the main source code, as it's a likely stumbling block for anyone who comes along and tries to remove it.

I prefer the first option, but I'm open to other alternatives as well.

Reply:
```
const mockExit = jest.spyOn(process, 'exit').mockImplementation((number) => {
  throw new Error(`process.exit(${number})`)
})
```

I think this is a great option. In fact I think you and I both copied the same SO post or wherever I found that exact code when I needed it a couple of days ago 😉

Reply:
haha, great